vmdb/app/models/host_openstack_infra.rb

```diff
@@ -27,26 +27,29 @@ def ssh_users_and_passwords
   end
 
   def get_parent_keypair(type = nil)
-    self.ext_management_system.try(:authentication_best_fit, type)
+    # Get private key defined on Provider level, in the case all hosts has the same user
+    self.ext_management_system.try(:authentication_type, type)
   end
 
-  def auth_user_keypair(type = nil)
-    # HostOpenstackInfra is using auth key set on ext_management_system level, not individual hosts
-    cred = self.get_parent_keypair(type)
-    return nil if cred.nil? || cred.userid.blank?
-    [cred.userid, cred.auth_key]
+  def authentication_best_fit(type = nil)
```

Comment:
We might add default later, but we don't use it now. Or the default looks at least different here. It should not affect anything else than Infra Hosts, since all the specific stuff has been encapsulated here.